### PR TITLE
Add Game Engine Development

### PR DESCRIPTION
**Pull request by stevinz**
**https://github.com/stevinz/awesome-game-engine-dev**

**Resources geared toward development of high-level, fully featured game engines. For curious minds or those interested in game engine development.**

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- [Comment on 2481](https://github.com/sindresorhus/awesome/pull/2481#issuecomment-1376649472)
- [Comment on 2478](https://github.com/sindresorhus/awesome/pull/2478#issuecomment-1376658070)
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/main/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `add Swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The title of your entry should be title-cased and the URL to your list should end in `#readme`.
	- Example: `- [Software Architecture](https://github.com/simskij/awesome-software-architecture#readme) - The discipline of designing and building software.`
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it. **Instead use [#2242](https://github.com/sindresorhus/awesome/issues/2242) for incubation visibility**.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/main/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
	- Must not feature `Contributing` or `Footnotes` sections.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=main&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name, text, or a `Licence` section to the readme. GitHub already shows the license name and link to the full text at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/main/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
	- It can optionally be linked from the readme in a dedicated section titled `Contributing`, positioned at the top or bottom of the main content.
	- The section should not appear in the Table of Contents.
- All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a `Footnotes` section at the bottom of the readme. The section should not be present in the Table of Contents.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

### CODE REVIEW

**Comment by stevinz** [comment]
unicorn

**Review by LeoDog896 — CHANGES_REQUESTED** [review]
I *really* love the banner for your repository! Only things:
- Your CI is failing.
- Make sure your CI is integrated with [awesome-lint](https://github.com/sindresorhus/awesome-lint).

**Comment by stevinz** [comment]
@LeoDog896 Thanks for the complement!

I have added `awesome-lint` to the repo's GitHub [actions](https://github.com/stevinz/awesome-game-engine-dev/actions).

The CI is also running [`markdown-link-check`](https://github.com/gaurav-nelson/github-action-markdown-link-check), which occasionally throws errors on good links. I keep an eye on links it complains about, and generally update the list weekly. However, I know failing CI isn't a good look, so I have moved a few troublesome links to an archive file, [`ARCHIVE.md`](https://github.com/stevinz/awesome-game-engine-dev/blob/main/ARCHIVE.md).

Thanks for your feedback.

**Comment by AkaraDeshmukh03** [comment]
> I *really* love the banner for your repository! Only things:
> 
> - Your CI is failing.
> 
> - Make sure your CI is integrated with [awesome-lint](https://github.com/sindresorhus/awesome-lint).

Comment testing.

**Comment by vhladiienko** [comment]
> Your entry here should include a short description about the project/theme of the list. It should not describe the list itself.

Hi @stevinz, I think you need to rephrase the topic description a bit. I was going to suggest a few options, however after looking at the list, I do struggle a bit - your indeed awesome list has quite many items on **different** game development topic, rather than being focused on game **engine** development.

What I would also suggest is adding "back to top" links to every section of the list to help navigating between individual sections and contents. See [here](https://github.com/awesome-selfhosted/awesome-selfhosted), for example.

**Comment by stevinz** [comment]
@vhladiienko Thanks for taking time to look over my list and comment. Respectfully, I disagree with your opinion. While the list does indeed have many resources related to game development, the list is heavily focused on game engine development. These two topics have a large area of overlap. There are extensive engine related links, among them dozens of open source game engines, libraries related to GUI development, visual scripting, serialization, special effects shaders, cross platform compiling and more.

The list is rather long, maybe longer than @sindresorhus would prefer. However, I built this list over the past decade and have personally utilized every resource on this list in one way or another while working on several different game engines.

The "back to top" is a good idea, but goes against the linter by violating the "_double link_" rule. Also I feel that it clutters the look of the list. Originally I had another layer of indentation, which I preferred, but that violates the linter as well...

**Comment by vhladiienko** [comment]
@stevinz fully agree and support the effort, I might just use your list myself! I suppose, we might pass on the theme description guideline this time. 

Guess that self-hosted list got into the awesome list before the linter has been developed, then. Sorry to mislead you on this one! Great job!

**Review by Zolaika — APPROVED** [review]

**Comment by Karneades** [comment]
Hi @stevinz 

Wow, a really long list of resources! That's impressive.

A few comments
* Rewrite the description to not describe the list itself but the project/theme at the top of the readme. Therefore, rewrite "Specifically, this list is geared..." and "This list currently ..." 
* **All non-important but necessary content (like extra copyright notices, hyperlinks to sources, pointers to expansive content, etc) should be grouped in a Footnotes section at the bottom of the readme. The section should not be present in the Table of Contents.** => for me that point matches the "licsense legend".
* The table of content and the whole list is so long, I suggest to reduce the nested lists where possible. E.g. don't use the section name (e.g. Dart) and another bullet point to just describe something again (Dart: - General -). It would make the list shorter and easier to use for me.

**Comment by stevinz** [comment]
@Karneades

Thanks for taking the time to look over my list. I have made some changes taking into account the points you and others have brought up, including:

- Revised the description at the top of the list to be more concise and definitive.
- Moved the "License Legend" to the bottom of the list.
- Reduced / collapsed some redundant sections. Includes moving the _"General"_ and _"Awesome Collections"_ links directly under the section headers.
- Removed the non-standard octocat (:octocat:) emoji. An item's primary link now points to it's GitHub repository by default (when available). Any additional resources are appended to the item's description. i.e. [[Docs]() | [Website](), etc.]

**Comment by MartinZikmund** [comment]
I like the list, very extensive one!

- In the description you use "This would include" in the second paragraph. I would rephrase this as "This includes"
- In the **Contributing** section you start with non-capitalized "see", I would use "See" to be in line with all other content.
- In some cases, the description of the list items starts with parentheses - I think that violates the requirement that description should start with capital letter.

**Review by MartinZikmund — APPROVED** [review]

**Comment by stevinz** [comment]
@MartinZikmund 

Thank you for the review. Those are nice, detailed observations. I have made changes along the lines of your suggestions.
- Replaced "This would include"
- Capitalized "See" in _Contributing_
- Removed parentheses from the start of descriptions

**Review by tomtau — APPROVED** [review]
A nice list, but some of it is overlapping with Learn Gamedev a bit?

File: readme.md [hunk 1]
```diff
@@ -565,6 +565,7 @@
 - [Games of Coding](https://github.com/michelpereira/awesome-games-of-coding#readme) - Learn a programming language by making games.
 - [Esports](https://github.com/strift/awesome-esports#readme) - Video games played as a sport.
 - [Learn Gamedev](https://github.com/notpresident35/awesome-learn-gamedev#readme) - The craft of video game creation.
+- [Game Engine Development](https://github.com/stevinz/awesome-game-engine-dev#readme) - Building software to speed game creation.
```
Comment (sindresorhus) [inline readme.md:568]:
```suggestion
- [Game Engine Development](https://github.com/stevinz/awesome-game-engine-dev#readme) - Building software to speed up game creation.
```

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1671837998105739266

**Review by maanu1234 — COMMENTED** [review]
👍

**Comment by Davison33** [comment]
[image: PDF file]
Private_profile_with_photos_a…
<https://drive.google.com/file/d/19e9ZLGuipYyEqpUG06y-EinPwP7U-8yn/view?usp=drivesdk>
[image: PDF file]
BG Sign-In - Sheet1.pdf
<https://drive.google.com/file/d/1cZKBiQqAzl5wYgKYP5DTtoLcfAFjoHTj/view?usp=drivesdk>

**Review by maanu1234 — COMMENTED** [review]
👍
